noir_stdlib/src/collections/map.nr

```diff
@@ -253,6 +362,38 @@ where
             }
         }
     }
+
+    // Apply HashMap's hasher onto key to obtain pre-hash for probing.
+    fn hash(self, key: K) -> u64
+    where
+        K: Hash,
+        B: BuildHasher<H>,
+        H: Hasher
+    {
+        let mut hasher = self._build_hasher.build_hasher();
+        key.hash(&mut hasher);
+        hasher.finish() as u64
+    }
+
+    // Probing scheme: quadratic function.
+    // We use 0.5 constant near variadic attempt and attempt^2 monomials.
+    // This ensures good uniformity of distribution for table sizes
+    // equal to prime numbers or powers of two. 
+    fn quadratic_probe(self, hash: u64, attempt: u64) -> u64 {
+        (hash + (attempt + attempt * attempt) / 2 ) % N
+    }
+
+    // Amount of elements in the table in relation to available slots exceeds α_max. 
+    // To avoid division with potential incorrectness due to floating point values,
+    // we conduct cross-multiplication.
+    // n / m >= MAX_LOAD_FACTOR_NUMERATOR / MAX_LOAD_FACTOR_DEN0MINATOR 
+    // n * MAX_LOAD_FACTOR_NUMERATOR >= m * MAX_LOAD_FACTOR_DEN0MINATOR
```

Comment:
Shouldn't this be `n * MAX_LOAD_FACTOR_DEN0MINATOR >= m * MAX_LOAD_FACTOR_NUMERATOR`?

Reply:
Indeed, thank you for noticing